lms/templates/discussion/_user_profile.html

```diff
@@ -3,7 +3,8 @@
 <div class="user-profile">
   <div class="sidebar-username"><a class="leaner-profile-link" href="${learner_profile_page_url}">${django_user.username | h}</a></div>
   <div class="sidebar-user-roles">
-      ${_(', ').join(sorted(set(map(_, [role.name for role in django_user.roles.all()]))))}
+    <% role_names = django_user.roles.order_by("name").values_list("name", flat=True).distinct() %>
+    ${", ".join(_(role_name) for role_name in role_names)}
```

Comment:
I haven't done any local verification -- I just rewrote the line while consulting the documentation. Part of the problem is that I don't know what data I would need to manually test this.